### PR TITLE
Fix `self._application_commands.pop` in `remove_application_command`

### LINKED ISSUE #1390: `bot.remove_application_command` doesn't clear command remnants correctly
### Summary

bot.remove_application_command doesn't clear command remnants correctly

### Reproduction Steps

- Load any cog with application command
- Unload this cog 

### Minimal Reproducible Code

```python
# main.py
import discord

intents = discord.Intents.All()

client = discord.Bot(intents=intents)
client.load_extension('Tests')
client.run('TOKEN')

#somewhere e.g. in command
client.unload_extension('Tests')
```

```python
# Tests.py
from discord.ext import commands
import discord

class Tests(commands.Cog):

    def __init__(self, client):
        self.client = client
   
    @commands.slash_command(name="testing")
    @commands.is_owner()
    async def testing(self, ctx: discord.ApplicationContext, member: discord.Member = None):
        await ctx.respond("test")

def setup(client):
    client.add_cog(Tests(client))
```

### Expected Results

bot.remove_application_command executed during cog unloading should pop command from self._application_commands correctly:

image

### Actual Results

bot.remove_application_command doesn't pop command from self._application_commands

image

### Intents

discord.Intents.all()

### System Information

```
- Python v3.8.13-final
- py-cord v2.0.0-candidate
    - py-cord pkg_resources: v2.0.0rc1
- aiohttp v3.8.1
- system info: Linux 5.4.0-107-generic #121-Ubuntu SMP Thu Mar 24 16:04:27 UTC 2022
```

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by yoggys**
## Summary

`self._application_commands` key types are strings, `.pop(int(command.id), None)` method doesn't affect any element of dict.
Change to `.pop(command.id, None)` will successfully remove the command from dict. This fixes issue #1390

image

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/bot.py [hunk 1]
```diff
@@ -165,8 +165,8 @@ def remove_application_command(self, command: ApplicationCommand) -> Optional[Ap
             except ValueError:
                 return None
             return self._pending_application_commands.pop(index)
-
-        return self._application_commands.pop(int(command.id), None)
+
```
Comment (Lulalaby) [inline discord/bot.py:168]: Unnecessary spaces

**Review by Lulalaby — CHANGES_REQUESTED** [review]
And please use a descriptive title instead of Fixes #xyz. 

Instead write this into the description

File: discord/bot.py (hunk 1 above)
Comment (yoggys) [reply discord/bot.py]: Sure, I edited it

**Review by Lulalaby — APPROVED** [review]
Thank you! ❤️

**Review by Dorukyum — APPROVED** [review]

**Pull request merged**